WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java

```diff
@@ -2580,88 +2572,45 @@ private void trackTagLoaded(ReaderTag tag) {
-//        }
-//
-//        Context context = view.getContext();
-//        final ListPopupWindow listPopup = new ListPopupWindow(context);
-//        listPopup.setWidth(context.getResources().getDimensionPixelSize(R.dimen.menu_item_width));
-//        listPopup.setAdapter(new ReaderMenuAdapter(context, menuItems));
-//        listPopup.setDropDownGravity(Gravity.END);
-//        listPopup.setAnchorView(view);
-//        listPopup.setModal(true);
-//        listPopup.setOnItemClickListener(new AdapterView.OnItemClickListener() {
-//            @Override
-//            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
-//                if (!isAdded()) {
-//                    return;
-//                }
-//
-//                listPopup.dismiss();
-//                switch ((int) id) {
-//                    case ReaderMenuAdapter.ITEM_FOLLOW:
-//                        onFollowTapped(getView(), post.getBlogName(), post.blogId);
-//                        toggleFollowStatusForPost(post);
-//                        break;
-//                    case ReaderMenuAdapter.ITEM_UNFOLLOW:
-//                        onFollowingTapped();
-//                        toggleFollowStatusForPost(post);
-//                        break;
-//                    case ReaderMenuAdapter.ITEM_BLOCK:
-//                        blockBlogForPost(post);
-//                        break;
-//                    case ReaderMenuAdapter.ITEM_NOTIFICATIONS_OFF:
-//                        AnalyticsUtils.trackWithSiteId(Stat.FOLLOWED_BLOG_NOTIFICATIONS_READER_MENU_OFF, post.blogId);
-//                        ReaderBlogTable.setNotificationsEnabledByBlogId(post.blogId, false);
-//                        updateSubscription(SubscriptionAction.DELETE, post.blogId);
-//                        break;
-//                    case ReaderMenuAdapter.ITEM_NOTIFICATIONS_ON:
-//                        AnalyticsUtils.trackWithSiteId(Stat.FOLLOWED_BLOG_NOTIFICATIONS_READER_MENU_ON, post.blogId);
-//                        ReaderBlogTable.setNotificationsEnabledByBlogId(post.blogId, true);
-//                        updateSubscription(SubscriptionAction.NEW, post.blogId);
-//                        break;
-//                    case ReaderMenuAdapter.ITEM_SHARE:
-//                        AnalyticsUtils.trackWithSiteId(Stat.SHARED_ITEM_READER, post.blogId);
-//                        sharePost(post);
-//                        break;
-//                    case ReaderMenuAdapter.ITEM_VISIT:
-//                        AnalyticsTracker.track(Stat.READER_ARTICLE_VISITED);
-//                        ReaderActivityLauncher.openPost(view.getContext(), post);
-//                        break;
-//                }
-//            }
-//        });
-//        listPopup.show();
+    public void onButtonClicked(ReaderPost post, ReaderPostCardActionType actionType) {
+        switch (actionType) {
+            case FOLLOW:
```

Comment:
These actions are mostly copied from the commented out code (OnItemClickListener) above.